artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerReceiverContext.java

```diff
@@ -146,11 +146,14 @@ private RoutingType getRoutingType(Symbol[] symbols) {
    public void onMessage(Delivery delivery) throws ActiveMQAMQPException {
       Receiver receiver;
       try {
-         receiver = ((Receiver) delivery.getLink());
 
          if (!delivery.isReadable()) {
             return;
          }
+         if (delivery.isPartial()) {
```

Comment:
For posterity, the duplicate check has been removed, via NO-JIRA commit 86779195fa4d88ee7fce2a5408b294e1cffd9bcc rather than against ARTEMIS-1056 like the original.